### PR TITLE
Extract CVE from the references.

### PR DESCRIPTION
**Pull request by amine3**
Extract CVE from the references.

### CODE REVIEW

File: agent/osv_output_handler.py [hunk 1]
```diff
@@ -151,3 +156,14 @@ def calculate_risk_rating(risk_ratings: list[str]) -> str:
...
             return rating
     return "POTENTIALLY"
+
+
+def _extract_cve_reference_advisory(
+    references: List[Dict[str, str]]
+) -> str | List[str]:
+    for reference in references:
+        if reference["type"] == "ADVISORY":
+            cve_match = re.match(CVE_PATTERN, reference["url"])
```
Comment (3asm) [inline agent/osv_output_handler.py:166]: Should be case-insensitive.

File: tests/conftest.py [hunk 2]
```diff
@@ -168,6 +168,18 @@ def osv_output_missing_cve() -> str:
     return data
 
 
+@pytest.fixture(name="nvd_output")
+def nvd_output() -> str:
```
Comment (3asm) [inline tests/conftest.py:172]: Where is the unit test for the CVE in the reference?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: tests/conftest.py (hunk 2 above)
Comment (amine3) [reply tests/conftest.py:172]: It is there

File: agent/osv_output_handler.py [hunk 3]
```diff
@@ -18,6 +20,7 @@
     "MEDIUM": agent_report_vulnerability_mixin.RiskRating.MEDIUM,
     "HIGH": agent_report_vulnerability_mixin.RiskRating.HIGH,
 }
+CVE_PATTERN = r".*/(CVE-.*-.*)"
```
Comment (3asm) [inline agent/osv_output_handler.py:23]:
```suggestion
CVE_PATTERN = r".*/(CVE-[0-9]+-[0-9]+)"
```

**Review by 3asm — APPROVED** [review]

**Pull request merged**